File: lab1/RPS_java/app/src/main/java/com/example/rps/Figure.java
package com.example.rps;



public class Figure {
    private String value;
    private String beats;


    public Figure (String value, String beats) {
        this.value = value;
        this.beats = beats;
    }

    public String getValue() {
        return this.value;
    }

    public int hierarchy (String figure) {
        if (figure.equals(this.beats))
            return 1;

        if (figure.equals(this.value))
            return 0;

        return -1;
    }
}

File: lab1/RPS_kotlin/app/src/main/java/com/example/rps/Figure.kt
package com.example.rps

import android.media.Image
import android.widget.ImageView

class Figure (value: String, beats: String) {
    private var value: String? = null
    private var beats: String? = null

    init {
        this.value = value
        this.beats = beats
    }

    fun getValue (): String? {
        return this.value
    }

    fun hierarchy (node: String): Int {
        if (node == this.beats)
            return 1

        if (node == this.value)
            return 0

        return -1
    }
}